app/controllers/shipit/api/tasks_controller.rb

```diff
@@ -23,6 +23,23 @@ def trigger
           message: 'A task is already running.',
         })
       end
+
+      def abort
+        if task.active?
+          task.abort!(aborted_by: current_user)
+          head(:accepted)
+        else
+          render(status: :method_not_allowed, json: {
```

Comment:
My thinking around making this a `method_not_allowed` was to surface to the user that the task may have already finished and therefore can't still be aborted (so in that case, I don't think the user would need to retry this action). Let me know if I've misunderstood though!

Reply:
I would use the `not_acceptable` (406 Not Acceptable) response code, since the method is in fact allowed, just no longer at the moment. I don't think a 200 response code is good enough, though. It would differ from the expected response (202), sure, but does not convey the actual reason why the response is different.

Reply:
Okay, I went and learned some more about different response codes. 406 seems like it's meant to be pretty specifically about incorrect headers, so at this point I'm thinking maybe even a 422 makes the most sense.

Reply:
I would maybe suggest [409 Conflict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409)? It looks like the "reason" this request would error is that the requested task is actually not actually active (which is not a correct state to apply the `abort` action to).

I would also like us to rethink the possibility of a success code rather than a response? If the user-requested action is to apply the `abort` operation and the task is not actually active, that feels like a success to me? possibly `204 - No Content`? This feels like the same discussion for calling a `DELETE` on a non-existent resource -- do you return "success" because the resource does not exist, or do you return an error because you were not able to "delete the resource"?

I would argue the meaning behind what we want to convey with this status code and what the user really cares about. 

- Does the user only care that the task is not actually running? I would argue a success code in this case. 
- Does the user want to ensure that an `abort` action was taken on a valid task? I would argue an error code.

Especially because this library is public, I would be cautious about exactly what we intend to convey in these situations.

_Sorry for this really long comment; I used to maintain a lot of REST APIs, so I am somewhat opinionated on things like this._

Reply:
No need to apologize! Having read up more on it, I think that if we do go for an error response, 409 does seem like the right choice to me.

I'll consider those questions (and others can chime in here too). But here's another variable to throw in this decision: the other reason I went for `method_not_allowed` is consistency -- we return `method_not_allowed` when you attempt to `request_merge!` on an already merged PR. To me, these are pretty analogous situations.

```
      def update
        merge_request = MergeRequest.request_merge!(stack, params[:id], current_user)
        if merge_request.waiting?
          head(:accepted)
        elsif merge_request.merged?
          render(status: :method_not_allowed, json: {
            message: "This pull request was already merged.",
          })
```